app/graphql/mutations/attach_files_to_sample.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Mutations
+  # Mutation that attaches files to sample
+  class AttachFilesToSample < BaseMutation
+    description 'Attaches files to sample.'
+    argument :files, [String], required: true, description: 'A list of files (signedBlobId) to attach to the sample'
+    argument :sample_id, ID,
+             required: true,
+             description: 'The Node ID of the sample to be updated.'
+
+    field :errors, [String], null: false, description: 'A list of errors that prevented the mutation.'
+    field :sample, Types::SampleType, null: false, description: 'The updated sample.'
+    field :status, GraphQL::Types::JSON, null: false, description: 'The status of the mutation.'
+
+    def resolve(sample_id:, files:)
+      sample = IridaSchema.object_from_id(sample_id, { expected_type: Sample })
+      files_attached = Attachments::CreateService.new(current_user, sample, { files: }).execute
+      {
+        sample:,
+        status: files_attached,
```

Comment:
**nitpick (blocking):** Can you change `status` to return a custom json object instead of the attachments. Since if you try and upload an attachment again to a sample you get back a weird response in status which doesn't quite make sense.

e.g.

```
{
  "data": {
    "attachFilesToSample": {
      "sample": {
        "id": "gid://irida/Sample/51"
      },
      "status": [
        {
          "id": null,
          "metadata": {},
          "deleted_at": null,
          "attachable_type": "Sample",
          "attachable_id": 51,
          "created_at": null,
          "updated_at": null,
          "log_data": null
        }
      ],
      "errors": []
    }
  }
}
```

and also `errors` is not populated.

If you look at `app/controllers/projects/samples/attachments_controller.rb` if sets a status and then in the view `app/views/projects/samples/attachments/create.turbo_stream.rb` it extracts the errors.

Similar logic should be done here to build up the status, and populate the errors.

I am not quite sure what `status` should look like, I had thoughts of keying it by `signed_blob_id` but that would take a bit of inspection into the `attachment` and not sure if one to one. In general I think `status` should have a code, 3 possible and potentially include the `attachments` created, though I think that should wait for `attachment` type. Maybe report what `signed_blob_ids` were successfully attached.

Reply:
Changes have been made.
I ended up using the signed_blob_id as the key for status value, and then made `:success` and `:error` as the status values. If you want me to change this let me know.

There is also a quirk with the `errors` returned value which MUST be an array, so right now errors output as 
```
[
  'signed--blob_id["error description"]`
]
```
Which is kind of ugly. Not sure what the best solution to pretty it would be.

Reply:
Let's change the `errors` to be a `JsonType` so that we can have readable error messages.

Reply:
Ah that should have been obvious to fix that lol. Changed now.